store/searchlayer/user_layer.go

```diff
@@ -39,7 +39,8 @@ func (s *SearchUserStore) Search(teamId, term string, options *model.UserSearchO
 				mlog.Error("Encountered error on Search.", mlog.String("search_engine", engine.GetName()), mlog.Err(err))
 				continue
 			}
-			if len(listOfAllowedChannels) == 0 {
+
```

Comment:
What was the root cause? I don't have enough context right now. Thank you!

Reply:
The idea is, listOfAllowedChannels can be 0 length if you have no channels allowed there, or can be nil if you have no restrictions about what channels you can see.